tests/grid/cell_id_04.cc

```diff
@@ -0,0 +1,91 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check CellId
+
+#include "../tests.h"
+
+#include <deal.II/base/geometry_info.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/distributed/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+
+#include <fstream>
+#include <sstream>
+
+using namespace dealii;
+
+template <int dim>
+void check (Triangulation<dim> &tr)
+{
+  typename Triangulation<dim>::cell_iterator cell = tr.begin(),
+                               endc = tr.end();
+
+
+  for (; cell!=endc; ++cell)
+    {
+      deallog << cell->level() << " " << cell->index() << std::endl;
+
+      // Store the CellId, convert it to a binary representation,
+      // create a new CellId from that, and create a cell iterator
+      // pointing to the same cell
+
+      const CellId cid = cell->id();
+
+      const CellId::binary_type cids = cid.to_binary<dim>();
+
+      const CellId cid2(cids);
+
+      typename Triangulation<dim>::cell_iterator cell2 = cid2.to_cell(tr);
```

Comment:
I suppose you expect `cell==cell2` here, right? If so, maybe also `Assert` that that's so.